src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -67,7 +67,35 @@ private JObject GetInvokeResult(byte[] script, Signers signers = null)
             JObject json = new JObject();
             json["script"] = Convert.ToBase64String(script);
             json["state"] = engine.State;
-            json["gasconsumed"] = engine.GasConsumed.ToString();
+            json["gasconsumed"] = new BigDecimal(engine.GasConsumed, NativeContract.GAS.Decimals).ToString();
+            json["exception"] = GetExceptionMessage(engine.FaultException);
+            try
+            {
+                json["stack"] = new JArray(engine.ResultStack.Select(p => p.ToJson()));
+            }
+            catch (InvalidOperationException)
+            {
+                json["stack"] = "error: recursive reference";
+            }
+            if (engine.State != VMState.FAULT)
+            {
+                ProcessInvokeWithWallet(json, signers);
+            }
+            return json;
+        }
+
+        private JObject GetVerificationResult(UInt160 scriptHash, Signers signers = null)
+        {
+            var snapshot = Blockchain.Singleton.GetSnapshot();
+            var contract = snapshot.Contracts.TryGet(scriptHash);
+            var engine = ApplicationEngine.Create(TriggerType.Verification, new Transaction(), snapshot);
+            var context = engine.LoadScript(contract.Script, CallFlags.None, contract.Manifest.Abi.GetMethod("verify").Offset);
+            engine.LoadScript(Array.Empty<byte>(), CallFlags.None);
```

Comment:
What's this for? It's a no-op effectively.

Reply:
I think instead it should respect and properly pass `args` to the `verify` method.

Reply:
seems like it was trying todo the `_init` but forgot a piece. Looks very similar to
https://github.com/neo-project/neo/blob/22f80a7ccfb00af15006569921ed87536cf3f857/src/neo/Wallets/Wallet.cs#L400-L404